pkg/machines/components/vm/vmMigrateDialog.jsx

```diff
@@ -0,0 +1,194 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2021 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from 'cockpit';
+import React, { useState } from 'react';
+import {
+    Button,
+    Checkbox,
+    Form,
+    FormGroup,
+    Modal,
+    Radio,
+    TextInput,
+    Tooltip
+} from '@patternfly/react-core';
+
+import { migrateToUri } from '../../libvirt-dbus.js';
+import { isEmpty, isObjectEmpty } from '../../helpers.js';
+import { ModalError } from 'cockpit-components-inline-notification.jsx';
+
+const _ = cockpit.gettext;
+
+const DestUriRow = ({ validationFailed, destUri, setDestUri }) => {
+    return (
+        <FormGroup label={_("Destination URI")} fieldId="dest-uri"
+                   id="dest-uri"
+                   validated={validationFailed.destUri ? "error" : "default"}>
+            <TextInput id='dest-uri-input'
+                       validated={validationFailed.destUri ? "error" : "default"}
+                       value={destUri}
+                       placeholder={cockpit.format(_("Example, $0"), "qemu+ssh://192.0.2.16/system")}
+                       onChange={setDestUri} />
+        </FormGroup>
+    );
+};
+
+const TypeRow = ({ type, setType, setUndefineSource, setStorage }) => {
+    return (
+        <FormGroup label={_("Migrated VM state")}
```

Comment:
@garrett should we maybe be consistent and use inline radio groups as we do in most of the other dialogs?

Reply:
Inline radios in the other forms are at the top and switch between different modes.

As this would be a toggle state of on/off, we could rework it to be a checkbox, however. Perhaps something like this:

☑ Run virtual machine after migration

After looking at the dialog again and thinking about it further, I think the order is wrong. Persistent stuff like storage should come first. Stateful things should come afterward.

So, this order, in other words:

1. Location URI
2. Storage
3. Delete original (also storage-related)
4. Run after migration

Reply:
virt-manager has the option * [ ] temporary move, which is a miz of our online/offline and undefine options. I have no preference over our way or what virt-manager exposed. I just thought that what they do, it's pretty nice and compact and good enough for first iteration. In addition, users are already used to that tool, so having similar dialog fields if they make sense is to the users' advantage I feel .